Ghidra/Features/Decompiler/src/decompile/cpp/prettyprint.cc

```diff
@@ -16,20 +16,20 @@
 #include "prettyprint.hh"
 #include "funcdata.hh"
 
-const char *EmitXml::highlight[] = { "color=\"keyword\"",
-					      "color=\"comment\"",
-					      "color=\"type\"",
-					      "color=\"funcname\"",
-					      "color=\"var\"",
-					      "color=\"const\"",
-					      "color=\"param\"",
-					      "color=\"global\"",
+const char *EmitXml::highlight[] = { " color=\"keyword\"",
+					      " color=\"comment\"",
+					      " color=\"type\"",
+					      " color=\"funcname\"",
+					      " color=\"var\"",
+					      " color=\"const\"",
+					      " color=\"param\"",
+					      " color=\"global\"",
```

Comment:
Most of the changes are straightforward. May you please explain why adding the space before color here helps reduce extra spaces?

Reply:
1.
```
void EmitXml::print(const char *str,syntax_highlight hl)
{
  *s << "<syntax " << highlight[(int4)hl] << '>';
  xml_escape(*s,str);
  *s << "</syntax>";
```
In the above original function in case of ( hl == no_color && *str != '\0' ) the output will be:
`<syntax >str</syntax>`
with an extra space after first **syntax**

in case of ( hl == no_color && *str == '\0' ) the output will be:
`<syntax ></syntax>`
with an aempty **syntax** tag and extra space after it.

2.
```
int4 EmitXml::beginStatement(const PcodeOp *op)
{
  *s << "<statement " << highlight[(int4)no_color];
  if (op != (const PcodeOp *)0)
    *s << " opref=\"0x" << hex << op->getTime() << "\">";
  else
```
In the above original function in case of ( op != (const PcodeOp *)0 ) the output will be:
`<statement  opref="...>`
with an extra space after **statement** tag and so on.